src/D2L.CodeStyle.Analyzers/Immutability/MutabilityInspector.cs

```diff
@@ -223,6 +223,33 @@ HashSet<ITypeSymbol> typeStack
 		) {
 			var typeParameter = symbol as ITypeParameterSymbol;
 
+			var hostType = hostSymbol as INamedTypeSymbol;

```

Comment:
This new code is a complete thought, and unrelated to the code below it in the same method.  Consider breaking it into it's own helper method.

Reply:
Ah, I was viewing it that the method didn't need the hostSymbol coerced in to a INamedTypeSymbol until I had added the logic in the routine below.  Otherwise it was kinda pointless.  Like, up until the logic is in, it's okay if the hostSymbol *isn't* a named type symbol.